Add Testerina Function Mocking and Object Mocking details (#825)
ballerina-platform/ballerina-dev-website: downloads/swan-lake-release-notes/swan-lake-preview1/RELEASE_NOTE.md

Introducing the Mocking API for object and function mocking.

#### Function Mocking

> `MockFunction` object added to handle function mocking. The `@test:MockFn` annotation attached to the MockFunction specifies the function to mock. MockFunction objects are defined as such.

Author: Updated

```
@test:MockFn {
    functionName : "<function_to_mock>"
}
test:MockFunction mockObj = new();
```

Function mocking is brought about by using the following functions

> - `test:when(mockObj)` is used to initialize the mocking capability within a particular test case

Reviewer: ```suggestion
- The `test:when(mockObj)` is used to initialize the mocking capability within a particular test case
```

> - Allows us to use the associated mocking functions like `call()`, `thenReturn()`, `withArguments()`

Reviewer: ```suggestion
- This allows you to use the associated mocking functions like `call()`, `thenReturn()`, and `withArguments()`.
```

#### Object Mocking

Object mocking enables controlling the values of member variables and the behavior of the member functions of an object. 

> - Introduced creating a test double, which provides an equivalent mock in place of the real object

Reviewer: ```suggestion
- Introduced the ability to create a test double, which provides an equivalent mock in place of the real object
```

What is test double? Is it "double test"?

> 

Reviewer: ```suggestion

```

Remove this new line between the bullet points.

> - Introduced capability of stubbing the member function or member variable

Reviewer: ```suggestion
- Introduced the capability of stubbing the member function or member variable
```
